Prevent falling through floor when leaving bed

CiaranG opened the pull request:
Without this change, leaving a bed which is on a floor one node high
makes you fall through the floor. Additionally, sometimes (presumably
due to network lag) you would still be in the bed, which you could no
longer see, leaving you no option but to log out.

CiaranG committed:
Prevent falling through floor when leaving bed

Without this change, leaving a bed which is on a floor one node high
makes you fall through the floor. Additionally, sometimes (presumably
due to network lag) you would still be in the bed, which you could no
longer see, leaving you no option but to log out.

gnanet commented:
I can confirm that the change works in multiplayer mode

minetestc55 (0.4.9-0ppa1~ubuntu12.04.1)

CiaranG committed:
A couple of fixes...

1. save spawn position when the player gets in the bed, not when
everybody in the world is in bed
2. restore position to where the player got into bed from (so
never into a wall as it could do before)
